plugins/module_utils/turbo/server.py

```diff
@@ -272,6 +273,8 @@ def _terminate(result):
 
                 await embedded_module.load()
                 try:
+                    os.environ.clear()
+                    os.environ.update(env)
```

Comment:
`os.environ` is a global variable. We can get two concurrent async execution of embedded_module. In this case, the second runs would overwrite the environment for the first calll. We should use a lock like in `load()` to protect to block.

Reply:
e.g:
```python3
#!/usr/bin/env python3

import asyncio
import os

async def my_async_loop(expectation):
    for _ in range(100):
        await asyncio.sleep(0.1)
        assert os.environ["baba"] == expectation

async def main():
    os.environ["baba"] = "loop1"
    l1 = my_async_loop("loop1")
    os.environ["blabla"] = "loop2"
    l2 = my_async_loop("loop2")
    await l2
    await l1

asyncio.run(main())
```

Reply:
I added a lock. I'll also mention two things. The `EmbeddedModule.run()` method needed a lock before my changes since it's modifying sys.stdin and then (possibly) awaiting the module. Also, unless I'm missing something, I don't think the lock in the `load()` or `unload()` methods is doing anything. It's not hurting anything to be there, but it's also not necessary. There's nowhere in either of those functions that a task switch can occur.